### PR TITLE
Database admin routes register without router

### PR DESCRIPTION
**Pull request by ChrisPdgn**
**What kind of change does this PR introduce?** 

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** 

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by kkopanidis — CHANGES_REQUESTED** [review]

File: modules/database/src/Database.ts [hunk 1]
```diff
@@ -296,7 +298,7 @@ export default class DatabaseModule extends ManagedModule<void> {
   async setSchemaExtension(call: CreateSchemaExtensionRequest, callback: SchemaResponse) {
     try {
       const schemaName = call.request.extension.name;
-      const extOwner = (call.metadata!.get('module-name')![0] as string) as string;
+      const extOwner = call.metadata!.get('module-name')![0] as string as string;
```
Comment (kon14) [inline modules/database/src/Database.ts:301]: Not sure why that double cast was previously there in the first place, but a single one should suffice.

File: modules/database/src/controllers/customEndpoints/customEndpoint.controller.ts [hunk 2]
```diff
@@ -44,8 +50,11 @@ export class CustomEndpointController {
           CustomEndpointHandler.addNewCustomOperationControl(schema);
         });
 
-        this.router.addRoutes(routes, false);
-        this.router.requestRefresh();
+        this.router!.addRoutes(routes, false);
+        this.router!.requestRefresh();
+      })
+      .then((r: any) => {
```
Comment (kon14) [inline modules/database/src/controllers/customEndpoints/customEndpoint.controller.ts:56]: Use `ICustomEndpoint` for this type.

File: modules/database/src/controllers/cms/schema.controller.ts [hunk 3]
```diff
@@ -30,7 +31,15 @@ export class SchemaController {
     });
   }
 
+  setRouter(router: DatabaseRoutes) {
+    this.router = router;
+    this.refreshRoutes();
+  }
+
   refreshRoutes() {
+    if (this.router === null) {
```
Comment (kon14) [inline modules/database/src/controllers/cms/schema.controller.ts:40]: We probably shouldn't warn for these. Just return from the function.
Same for rest of these checks.

**Review by kon14 — CHANGES_REQUESTED** [review]

File: modules/database/src/controllers/customEndpoints/customEndpoint.controller.ts (hunk 2 above)
Comment (kon14) [reply modules/database/src/controllers/customEndpoints/customEndpoint.controller.ts:56]: My bad, we don't actually need the response at all.

**Review by kon14 — CHANGES_REQUESTED** [review]

**Review by kkopanidis — APPROVED** [review]

**Review by kon14 — APPROVED** [review]

**Pull request merged**